### ISSUE TITLE
Add Json support to reduce need for so many key/value pairs

### ISSUE THREAD

**Issue by cgibbons** [aws/aws-dotnet-extensions-configuration#72]
It would be nice if this library supported a string that contained JSON and would parse that string and create settings based on that string, exactly how the standard appsettings.json works.

## Describe the Feature
Most developers in .net core are used to storing their settings in appsetting.json files in a json format, it would be nice if a single key could contain a full json string that could parsed into configuration keys in the same manner.

## Proposed Solution
It looks like most the work is nearly done since there is already a json parser for using the secrets manager

---

This is a :rocket: Feature Request

**Comment by cgibbons** [comment]
It looks like there is already a PR waiting to support this.  

https://github.com/aws/aws-dotnet-extensions-configuration/pull/69

**Comment by mnisha** [comment]
Is this option available?
How can I parse the value as json when using AddSystemsManager ?

**Comment by cgibbons** [comment]
Is there any update on this issue?  The completed pull request has been pending since June 2020.

**Comment by esmelov** [comment]
is there anyone alive?

**Comment by arturomonge** [comment]
I was also waiting on this, but with the release of AWS AppConfig support (https://github.com/aws/aws-dotnet-extensions-configuration/pull/82) I've started moving my SSM Params configs to AppConfig and so far it works nicely to solve this issue. I can now manage my configs using JSON instead of tons of individual SSM params. 

IMO, now that it's supported by the .NET config provider and now that I've started using it, I think AppConfig is a much better choice for managing configurations, instead of SSM Params. It has environments, profiles (e.g. different configs for different projects in a solution), validation, etc. And you can just work with production configs as you do with `appsettings.json`. It comes with an added cost, true, but in my case, not having to deal with the hassle of managing individual SSM Params and struggling with some configs is definitely worth it.

Support for AppConfig probably solves the needs described in these PRs too: https://github.com/aws/aws-dotnet-extensions-configuration/pull/69 https://github.com/aws/aws-dotnet-extensions-configuration/issues/51 https://github.com/aws/aws-dotnet-extensions-configuration/pull/54

I don't think the new AppConfig support works with the Lambda AppConfig extension, so for Lambda functions, I'm sticking to SSM params for the time being. Without support for the extension, AppConfig would become too expensive. But for ASP.NET Core APIs in ECS tasks, it's definitely worth taking a look at this.

**Comment by gbd3-en** [comment]
This functionality has been implemented in [master](https://github.com/aws/aws-dotnet-extensions-configuration/pull/103).   We are successfully using that to pull multiple ASP.NET Core configuration values from JSON in a single SSM parameter.

> normj commented [on Apr 14](https://github.com/aws/aws-dotnet-extensions-configuration/pull/103#issuecomment-1098760287)
>Merged into master and working on getting master released.

Are there any updates on when v4 might release?

**Comment by ashishdhingra** [comment]
> This functionality has been implemented in [master](https://github.com/aws/aws-dotnet-extensions-configuration/pull/103). We are successfully using that to pull multiple ASP.NET Core configuration values from JSON in a single SSM parameter.
> 
> > normj commented [on Apr 14](https://github.com/aws/aws-dotnet-extensions-configuration/pull/103#issuecomment-1098760287)
> > Merged into master and working on getting master released.
> 
> Are there any updates on when v4 might release?

@cgibbons / @gbd3-en The [Amazon.Extensions.Configuration.SystemsManager 4.0.0](https://www.nuget.org/packages/Amazon.Extensions.Configuration.SystemsManager/4.0.0) is already out there. Hence closing this issue.
